tests/src/socket_test.rs

```diff
@@ -0,0 +1,96 @@
+
+use std::future::Future;
+
+use http::{Request, header, Method, Response, StatusCode};
+use hyper::{Body, upgrade::Upgraded};
+use rand::RngCore;
+use shared::{http_client::SamplyHttpClient, MsgId, MsgEmpty, beam_id::AppOrProxyId};
+use tokio::io::{AsyncWriteExt, AsyncReadExt};
+use anyhow::Result;
+use crate::*;
+
+async fn upgrade(res: impl Future<Output = Response<Body>>) -> Upgraded {
+    hyper::upgrade::on(res.await).await.expect("Upgrade successfull")
+}
+
+async fn test_connections(r1: impl Future<Output = Response<Body>>, r2: impl Future<Output = Response<Body>>) {
+    let (mut a, mut b) = tokio::join!(
+        upgrade(r1),
+        upgrade(r2)
+    );
+    const N: usize = 2_usize.pow(13);
+    let test_data: &mut [u8; N] = &mut [0; N];
+    rand::thread_rng().fill_bytes(test_data);
+    let mut read_buf = [0; N];
+    a.write_all(test_data).await.unwrap();
+    a.flush().await.unwrap();
+    b.read_exact(&mut read_buf).await.unwrap();
+    assert_eq!(test_data, &read_buf);
+}
+
+#[tokio::test]
+async fn test_full() -> Result<()> {
+    let client = shared::http_client::build(&Vec::new(), None, None)?;
+    let client2 = client.clone();
+
+    // let res = post_socket_req(client.clone(), task_id).await?;
+    // assert_eq!(res.status(), StatusCode::CREATED);
```

Comment:
Please Remove dead code (however, in the tests this is not a blocker)
I will not comment this for the rest of the tests.

Reply:
I will still clean it up a bit